Patch-Source: https://github.com/beanz/anyevent-mocktcpserver-perl/pull/1.patch
--
From 7d8d077445d48eba4b2aeaf37e269e7a12b729b9 Mon Sep 17 00:00:00 2001
From: Graham Knop <haarg@haarg.org>
Date: Sun, 27 Aug 2023 13:04:29 +0200
Subject: [PATCH] remove unhandled import arguments

Previous versions of perl allow a use or import call even if the import
method doesn't exist. Future versions are intending to throw errors for
calls to an undefined import method that includes arguments.

Remove the arguments to use lines when the import method does not exist.
---
 t/01-simple.t     | 2 +-
 t/02-unexpected.t | 2 +-
 t/03-two.t        | 2 +-
 t/04-timeout.t    | 2 +-
 t/05-error.t      | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/t/01-simple.t b/t/01-simple.t
index 26651de..019dfbc 100644
--- a/t/01-simple.t
+++ b/t/01-simple.t
@@ -8,7 +8,7 @@ use constant {
 };
 use Test::More;
 use AnyEvent::Socket;
-use AnyEvent::MockTCPServer qw/:all/;
+use AnyEvent::MockTCPServer;
 
 my $done = AnyEvent->condvar;
 my $server;
diff --git a/t/02-unexpected.t b/t/02-unexpected.t
index c6a5c4e..95d098b 100644
--- a/t/02-unexpected.t
+++ b/t/02-unexpected.t
@@ -11,7 +11,7 @@ BEGIN {
 }
 use Test::More;
 use AnyEvent::Socket;
-use AnyEvent::MockTCPServer qw/:all/;
+use AnyEvent::MockTCPServer;
 
 my $done = AnyEvent->condvar;
 my $server;
diff --git a/t/03-two.t b/t/03-two.t
index 792f721..c2cedc3 100644
--- a/t/03-two.t
+++ b/t/03-two.t
@@ -8,7 +8,7 @@ use constant {
 };
 use Test::More;
 use AnyEvent::Socket;
-use AnyEvent::MockTCPServer qw/:all/;
+use AnyEvent::MockTCPServer;
 
 my $done = AnyEvent->condvar;
 my $done2 = AnyEvent->condvar;
diff --git a/t/04-timeout.t b/t/04-timeout.t
index 5e6bfd1..e5afe40 100644
--- a/t/04-timeout.t
+++ b/t/04-timeout.t
@@ -11,7 +11,7 @@ BEGIN {
 }
 use Test::More;
 use AnyEvent::Socket;
-use AnyEvent::MockTCPServer qw/:all/;
+use AnyEvent::MockTCPServer;
 
 my $done = AnyEvent->condvar;
 my $server;
diff --git a/t/05-error.t b/t/05-error.t
index 8f45798..98d3a30 100644
--- a/t/05-error.t
+++ b/t/05-error.t
@@ -11,7 +11,7 @@ BEGIN {
 }
 use Test::More;
 use AnyEvent::Socket;
-use AnyEvent::MockTCPServer qw/:all/;
+use AnyEvent::MockTCPServer;
 
 my $done = AnyEvent->condvar;
 my $server;
